Add surepetcare component

benleb opened the pull request:
## Description:
Add a basic component for the Sure Petcare cat and pet flaps. Most important things like lock state of a flap and location of pets is implemented, will eventually be enhanced if component gets accepted.

**Related issue (if applicable):** https://community.home-assistant.io/t/sureflap-connect-pet-door/41488/54

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** -- Follows if component/component code itself is accepted -- home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml
surepetcare:
  username: x@y.com
  password: v3rys3cr3t!
  household_id: 1337
  flaps: [{id: 2337, name: Flap}]
  pets: [{id: 3337, name: Pet}]
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [X] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

benleb committed:
add surepetcare component

benleb committed:
pin surepy version

benleb committed:
remove comment

benleb committed:
make pylint happy

benleb committed:
line length shuld be 79

benleb committed:
line length shuld be 79

benleb committed:
remove unnecessary typing which confuses travis

benleb committed:
use enums & better attributes mapping

benleb committed:
add missing blank line

benleb committed:
mvp commit, various changes and fixes

benleb committed:
honor max line length of 79

benleb commented:
any plans to drop py35 support any time soon? :>

benleb committed:
update enums & logging format

benleb committed:
add class doc strings

benleb committed:
update init fuction, small fixes

benleb committed:
update enums, logging format & init function

frenck commented:
⚠️ I was unable to find a documentation PR (matching this code change) in our [documentation repository](https://github.com/home-assistant/home-assistant.io).

Please, update the documentation and open a PR for it (be sure to create a documentation PR against the `next` branch) and link 🔗 this PR and the new documentation PR by mentioning the link to the PR's in both openings posts/descriptions.

🏷 I am adding the `docs-missing` label until this has been resolved.

benleb commented:
> ⚠️ I was unable to find a documentation PR (matching this code change) in our [documentation repository](https://github.com/home-assistant/home-assistant.io).
> 
> Please, update the documentation and open a PR for it (be sure to create a documentation PR against the `next` branch) and link 🔗 this PR and the new documentation PR by mentioning the link to the PR's in both openings posts/descriptions.
> 
> 🏷 I am adding the `docs-missing` label until this has been resolved.

Because of that I wrote "...Follows if component/component code itself is accepted...". Did you overlook it or do you mean "no review without documentation"?

patatman commented:
> > ⚠️ I was unable to find a documentation PR (matching this code change) in our [documentation repository](https://github.com/home-assistant/home-assistant.io).
> > Please, update the documentation and open a PR for it (be sure to create a documentation PR against the `next` branch) and link 🔗 this PR and the new documentation PR by mentioning the link to the PR's in both openings posts/descriptions.
> > 🏷 I am adding the `docs-missing` label until this has been resolved.
> 
> Because of that I wrote "...Follows if component/component code itself is accepted...". Did you overlook it or do you mean "no review without documentation"?
> 
> I choose this way as I wont spend time to write a documentation for something which maybe someone here does not like because of ##insert what you want##, sorry.

I'm willing to write some docs, since i'm going to purchase this flap in the near future.
@frenck Is the component sufficient enough the be able to merge? Once the docs are finished that is.

freekeys commented:
Happy to help with docs too!

richard-parker commented:
I get the point about missing docs; how do we break the deadlock here? I’m sure a lot of folks would love this integration; it seems a shame to hold up its release over what is probably 1-2 pages of docs at most. Can we help with that, since @benleb has written the code?

Aekschen commented:
Thank you for this!
I wanted to test this component yesterday. Couldn't get it to work yet.
Is it legit that this component is uses Python 3.6+ functionalities like f-strings when Hass only requires 3.5.3+ ? I'm just asking because this caused problems at my end.

balloob committed:
Bumped version to 0.94.0b0

amelchio committed:
Update pysonos to 0.0.14 (#24185)

Danielhiversen committed:
upgrade broadlink library, Use cryptography instead of pycryptodome (#24186)

Danielhiversen committed:
danielhiversen as codeowner for yr (#24189)

balloob committed:
No longer allow invalid slugs or extra keys (#24176)

* No longer allow slugs

* Lint

* Remove HASchema

* Lint

* Fix tests

balloob committed:
Dynamic panels (#24184)

* Don't require all panel urls to be registered

* Allow removing panels, fire event when panels updated

balloob committed:
Dynamic panels (#24184)

* Don't require all panel urls to be registered

* Allow removing panels, fire event when panels updated

pvizeli committed:
Update azure-pipelines.yml for check version (#24194)

pvizeli committed:
Update azure-pipelines.yml for check version (#24194)

balloob committed:
Updated frontend to 20190530.0

balloob committed:
Updated frontend to 20190530.0

Jc2k committed:
homekit_controller no longer logs with transient network errors causing crypto failures as it will auto recover (#24193)

Jc2k committed:
homekit_controller no longer logs with transient network errors causing crypto failures as it will auto recover (#24193)

Danielhiversen committed:
Update Tibber library (#24192)

frenck committed:
:pencil2: Corrects incorrect command in hassfest (#24188)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Fix ESPHome discovered when already exists (#24187)

* Fix ESPHome discovered when already exists

* Update .coveragerc

OttoWinter committed:
Fix ESPHome discovered when already exists (#24187)

* Fix ESPHome discovered when already exists

* Update .coveragerc

balloob committed:
Update hass-nabucasa (#24197)

balloob committed:
Update hass-nabucasa (#24197)

balloob committed:
Allow discovery flows to be discovered via zeroconf/ssdp (#24199)

balloob committed:
Allow discovery flows to be discovered via zeroconf/ssdp (#24199)

balloob committed:
Bumped version to 0.94.0b1

balloob committed:
Improve error handling (#24204)

balloob committed:
Improve error handling (#24204)

vangorra committed:
Fixing tplink issues with offline devices during setup (#23668)

* Fixing tplink issues with offline devices during setup.

* Fixing circleci errors.

* Adding code to defer the creation of entities that are not online.

* Addressing code review changes and cleaning up a little.

* Fixing tests and static analysis.

* Adding test to satisfy coverage requirements.

* Resolving merge conflicts.

* Fixing issue where lights don't appear in the integration page.

* Using pyHS100 properties for most sysinfo.
Addressing some PR feedback.

* Addressing some PR feedback.

* Better testing async_add_entities_retry
Testing for static dimmers.
Making greater use of conf constants.

* Fixing all static analysis issues.

* Adding non-blocking call for getting discovering devices.

* Address PR feedback

therve committed:
Bump oauthlib version (#24111)

* Bump oauthlib version

* Bump fitbit instead

* Update requirements

therve committed:
Bump oauthlib version (#24111)

* Bump oauthlib version

* Bump fitbit instead

* Update requirements

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Fix ESPHome config flow with invalid config entry (#24213)

OttoWinter committed:
Fix ESPHome config flow with invalid config entry (#24213)

pvizeli committed:
Don't follow redirect on ingress itself (#24218)

* Don't follow redirect on ingress itself

* Fix comment

pvizeli committed:
Don't follow redirect on ingress itself (#24218)

* Don't follow redirect on ingress itself

* Fix comment

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Instantiate lock inside event loop (#24203)

balloob committed:
Instantiate lock inside event loop (#24203)

balloob committed:
Use resource for index routing. (#24223)

balloob committed:
Use resource for index routing. (#24223)

Kane610 committed:
Axis - Handle Vapix error messages (#24215)

Kane610 committed:
Axis - Handle Vapix error messages (#24215)

balloob committed:
Add manifest support for homekit discovery (#24225)

* Add manifest support for homekit discovery

* Add a space after model check

* Update comment

balloob committed:
Add manifest support for homekit discovery (#24225)

* Add manifest support for homekit discovery

* Add a space after model check

* Update comment

pnbruckner committed:
Add control of Amcrest indicator light (#23986)

Enable feature by default but allow it to be disabled by "control_light: false" in config.

Get brand from camera instead of assuming Amcrest (since this works with other cameras, too.)

Retrieve RTSP URL in update method instead of in stream_source property and in handle_async_mjpeg_stream method.

Move amcrest imports from methods to global.

balloob committed:
Bumped version to 0.94.0b2

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Adminiuga committed:
ZHA requirements version bump. (#24228)

* ZHA requirements version bump.
* zha-quirks version bump.

Kane610 committed:
Don't allow more than one config flow per discovered Axis device (#24230)

Kane610 committed:
Don't allow more than one config flow per discovered Axis device (#24230)

soldag committed:
Bump pychromecast (#24234)

balloob committed:
Log HomeKit model (#24229)

balloob committed:
Log HomeKit model (#24229)

balloob committed:
Add GPSLogger device_info and unique_id (#24231)

balloob committed:
Add GPSLogger device_info and unique_id (#24231)

balloob committed:
GeoFency unique ID and device info (#24232)

balloob committed:
GeoFency unique ID and device info (#24232)

rytilahti committed:
add a deprecation warning for tplink device_tracker (#24236)

* add a deprecation warning for tplink device_tracker

* reword the warning a bit

rytilahti committed:
add a deprecation warning for tplink device_tracker (#24236)

* add a deprecation warning for tplink device_tracker

* reword the warning a bit

balloob committed:
Mobile app to use device tracker config entry (#24238)

* Mobile app to use device tracker config entry

* Lint

* Re-use device_info

* Lint

balloob committed:
Mobile app to use device tracker config entry (#24238)

* Mobile app to use device tracker config entry

* Lint

* Re-use device_info

* Lint

balloob committed:
Update home zone when core config updated (#24237)

* Update home zone when core config updated

* Lint

balloob committed:
Bumped version to 0.94.0b3

balloob committed:
Do not use the cache dir for PIP installs (#24233)

balloob committed:
Do not use the cache dir for PIP installs (#24233)

danielsjf committed:
Round Awair sensor values (#24093)

* Round sensor values

* Add code owner

* Update code owners

* Fix tests

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

kbickar committed:
Sense update (#24220)

* Changed updates to be done by component and updates dispatched

* syntax updates

* Added code owner

* Removed whitespace

* Updated CODEOWNERS

* Added subscription undoer

Emilv2 committed:
fix gitlab_ci sad icon (#24241)

austinmroczek committed:
Add details to triggered state for total connect alarms (#24106)

* Bump skybellpy to 0.4.0

* Bump skybellpy to 0.4.0 in requirements_all.txt

* Added extra states for STATE_ALARM_TRIGGERED to allow users to know if
it is a burglar or fire or carbon monoxide so automations can take
appropriate actions.  Updated TotalConnect component to handle these new
states.

* Fix const import

* Fix const import

* Fix const imports

* Bump total-connect-client to 0.26.

* Catch details of alarm trigger in state attributes.

Also bumps total_connect_client to 0.27.

* Change state_attributes() to device_state_attributes()

cereal2nd committed:
Bump python-velbus version for velbus component (#24226)

* Bump python-velbus version

* Bump python velbus version

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Updated frontend to 20190601.0

balloob committed:
Updated frontend to 20190601.0

balloob committed:
Bumped version to 0.94.0b4

frenck committed:
Adds AdGuard Home integration (#24219)

* Adds AdGuard Home integration

* :shirt: Addresses linting warnings

* :ambulance: Fixes typehint in async_setup_entry

* :shirt: Take advantage of Python's coalescing operators

* :shirt: Use adguard instance from outer scope directly in service calls

* :shirt: Use more sensible scan_interval default for sensors

* :shirt: Adds specific files to .coveragerc

* :umbrella: Added tests and small changes to improve coverage

* :hammer: Import adguardhome dependencies at the top

* :ambulance: Converted service handlers to be async

* :fire: Removed init step from config flow

Emilv2 committed:
Fix typo in integration component (#24250)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Kane610 committed:
Replace pyunifi with aiounifi in UniFi device tracker (#24149)

* Replace pyunifi with aiounifi

* Fix tests

* Add sslcontext

* Fix tests

* Fix import order

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Updated frontend to 20190602.0

balloob committed:
Updated frontend to 20190602.0

balloob committed:
Add restore state to OwnTracks device tracker (#24256)

* Add restore state to OwnTracks device tracker

* Lint

* Also store entity devices

* Update test_device_tracker.py

balloob committed:
Add restore state to OwnTracks device tracker (#24256)

* Add restore state to OwnTracks device tracker

* Lint

* Also store entity devices

* Update test_device_tracker.py

isabellaalstrom committed:
Add service for adding event to google component (#22473)

* Add service for adding event to google component

* Add service examples for google.add_event

* add refactoring based on reviews

* Fix too long lines

* Order import

* Move to move line

* Remove parenthesis

* Add service for adding event to google component

* Add service examples for google.add_event

* add refactoring based on reviews

* Add check for correct scopes, otherwise re-authenticate

* fix build failure

* fix build failure

balloob committed:
Add restore state to Geofency (#24268)

* Add restore state to Geofency

* Lint

balloob committed:
Add restore state to Geofency (#24268)

* Add restore state to Geofency

* Lint

balloob committed:
Mobile app device tracker to restore state (#24266)

balloob committed:
Mobile app device tracker to restore state (#24266)

pvizeli committed:
Bumped version to 0.94.0b5

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

OttoWinter committed:
Bump aioesphomeapi to 2.1.0 (#24278)

* Bump aioesphomeapi to 2.1.0

* Update requirements txt

OttoWinter committed:
Bump aioesphomeapi to 2.1.0 (#24278)

* Bump aioesphomeapi to 2.1.0

* Update requirements txt

fabaff committed:
Remove icon() (#24280)

Kane610 committed:
deCONZ migrate to SSDP discovery (#24252)

* Migrate deCONZ to use new SSDP discovery
Add new discovery info manufacturer URL to be able to separate Hue and deCONZ bridges

* Mark deCONZ as migrated in Discovery component

* Fix tests

* Fix Hue discovery ignore deCONZ bridge

* Less snake more badger

* Mushroom

* Fix indentation

* Config flow ignore manufacturer url that is not philips

Kane610 committed:
deCONZ migrate to SSDP discovery (#24252)

* Migrate deCONZ to use new SSDP discovery
Add new discovery info manufacturer URL to be able to separate Hue and deCONZ bridges

* Mark deCONZ as migrated in Discovery component

* Fix tests

* Fix Hue discovery ignore deCONZ bridge

* Less snake more badger

* Mushroom

* Fix indentation

* Config flow ignore manufacturer url that is not philips

Swamp-Ig committed:
Split devices of nodes with multiple instances (#24032)

* Split devices of nodes with multiple instances

* Note entities to register with device registry

* Use EntityPlatform

balloob committed:
Add a discovery config flow to Wemo (#24208)

balloob committed:
Add a discovery config flow to Wemo (#24208)

balloob committed:
Fix cors on the index view (#24283)

balloob committed:
Fix cors on the index view (#24283)

balloob committed:
Remove deps folder in config when on Docker (#24284)

* Remove deps folder in config

* Fix tests

* Fix tests with docker check

balloob committed:
Remove deps folder in config when on Docker (#24284)

* Remove deps folder in config

* Fix tests

* Fix tests with docker check

piitaya committed:
Add temperature sensor support to google smarthome thermostat device (#24264)

* Add temperature sensor support to google smarthome thermostat device

* fix lint for trait_test

* Reset temperature unit in tests

* Address comment

piitaya committed:
Add temperature sensor support to google smarthome thermostat device (#24264)

* Add temperature sensor support to google smarthome thermostat device

* fix lint for trait_test

* Reset temperature unit in tests

* Address comment

balloob committed:
Bumped version to 0.94.0b6

pvizeli committed:
Bumped version to 0.94.0b7

brandond committed:
Update pyvera to 0.3.1 for alert support (#24289)

balloob committed:
Updated frontend to 20190604.0

balloob committed:
Updated frontend to 20190604.0

balloob committed:
Update translations

balloob committed:
Guard against bad states in Mobile App/OwnTracks (#24292)

balloob committed:
Guard against bad states in Mobile App/OwnTracks (#24292)

balloob committed:
Run SSDP discovery in parallel (#24299)

balloob committed:
Run SSDP discovery in parallel (#24299)

pvizeli committed:
Create progress file for pip installs (#24297)

* Create progress file for pip installs

* fix dedlock

* unflacky test

* Address comments

* Lint

* Types

pvizeli committed:
Create progress file for pip installs (#24297)

* Create progress file for pip installs

* fix dedlock

* unflacky test

* Address comments

* Lint

* Types

balloob committed:
Upgrade Zeroconf to 0.23 (#24300)

balloob committed:
Upgrade Zeroconf to 0.23 (#24300)

balloob committed:
Bumped version to 0.94.0b8

gibman committed:
Expose specific device_class for Velux covers (#24279)

* Update cover.py

Blinds, Rollingshutters and Awnings did not set their respective device_class attribute
Previously they would all appear as device_class "window"

* fallback device class is always 'window'

fallback device class is always 'window' in the event we have an unknown cover type

* trailing whitespace removed, trimmed as well

* Update cover.py

balloob committed:
Fix OwnTracks race condition (#24303)

* Fix OwnTracks race condition

* Lint

balloob committed:
Fix OwnTracks race condition (#24303)

* Fix OwnTracks race condition

* Lint

Kane610 committed:
address is deprecated in favor of addresses (#24302)

Kane610 committed:
address is deprecated in favor of addresses (#24302)

balloob committed:
Merge remote-tracking branch 'origin/master' into rc

balloob committed:
Bumped version to 0.94.0

rolfberkenbosch committed:
Change meteoalertapi to version 0.1.3 (#24307)

* Change meteoalertapi to version 0.1.3

* Add requirements_all.txt file

pvizeli committed:
Create azure-pipelines-wheels.yml

pvizeli committed:
Update and rename azure-pipelines.yml to azure-pipelines-release.yml

pvizeli committed:
Rename azure-pipelines-release.yml to azure-pipelines.yml

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Rename azure-pipelines.yml to azure-pipelines-release.yml

pvizeli committed:
Update and rename azure-pipelines.yml to azure-pipelines-release.yml

scop committed:
Upgrade pytest and -cov (#24258)

* Upgrade pytest to 4.5.0

https://docs.pytest.org/en/latest/changelog.html#pytest-4-5-0-2019-05-11
https://docs.pytest.org/en/latest/changelog.html#pytest-4-4-2-2019-05-08

* Upgrade pytest to 4.6.0

https://docs.pytest.org/en/latest/changelog.html#pytest-4-6-0-2019-05-31

* Upgrade pytest-cov to 2.7.1

https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst#271-2019-05-03
https://github.com/pytest-dev/pytest-cov/blob/master/CHANGELOG.rst#270-2019-05-03

* Upgrade pytest to 4.6.1

https://docs.pytest.org/en/latest/changelog.html#pytest-4-6-1-2019-06-02

alengwenus committed:
Add LCN scene platform (#24242)

felipediel committed:
Add support for learning new commands (#23888)

* Add support for learning new commands

This update creates a generic service in the 'remote' component to enable remote control platforms to learn new commands.

* Update __init__.py with the proposed changes

- Add 'supported_features' property and a constant related to the 'learn_command' functionality.
- Redefine 'async_learn_command' function as a coroutine.

* Update __init__.py

* Fix assertion error

Adding the 'supported_features' attribute generated an assertion error on the 'Demo Remote' platform. This update fixes this.

* Fix duplicated 'hass' object

This update fixes a typo that occurred at the last update.

aequitas committed:
Upgrade to newer version of rflink with improve error handling on incoming data. (#24263)

Related: https://github.com/home-assistant/home-assistant/issues/23942

drobtravels committed:
SolarEdge Local Component (#23996)

* Basic local SolarEdge monitoring for energy / power

* Basic local SolarEdge monitoring for energy / power

* generated CODEOWNERS, requirements, excluded tests

* generated CODEOWNERS, requirements, excluded tests

* lint fixes, etc

* lint fixes, etc

* Fix docstyle for init

Of course thats the file I forgot to run tests on

* Load all sensors by default

They use the same API endpoint.  This changes was made per https://github.com/home-assistant/architecture/pull/244

* remve unneded date/time

* ran hassfest again

* add throttle when updating

* readd solax, mistakenly removed

* Update sensor.py

Kane610 committed:
Don't let zeroconf be smart with addresses (#24321)

Kane610 committed:
Don't let zeroconf be smart with addresses (#24321)

scarface-4711 committed:
Push to version 0.7.9 of denonavr (#24260)

pvizeli committed:
Merge pull request #24305 from home-assistant/rc

0.94.0

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-release.yml for Azure Pipelines

bachya committed:
Bump simplipy to 3.4.2 (#24326)

* Bump simplipy to 3.4.2

* Updated requirements

victorcerutti committed:
Meteofrance fix 24244 (#24315)

* Update meteofrance package version

Fix https://github.com/home-assistant/home-assistant/issues/24244

* Add code owner to manifest

* Update CODEOWNERS

jjlawren committed:
Only update media icon when necessary (#24324)

* Only update media icon when necessary

* Lint

* Comment

andre-richter committed:
components/cover: Typo in docstring. (#24329)

cgtobi committed:
Add exception handling for Netatmo climate (#24311)

* Add exception handling

* Make pylint happy

balloob committed:
Initiate websession inside event loop (#24331)

balloob committed:
Initiate websession inside event loop (#24331)

Commit:
Remove deprecated AlarmControlPanel (#24322)

dmulcahey committed:
Use node descriptor from Zigpy for ZHA (#24316)

* use zigpy node descriptor

* cleanup

bieniu committed:
Add abbreviation for light template variable names (#24336)

Commit:
Remove attribute lowBat (#24323)

davidbb committed:
Don't load last_checkpoint if shipment is pending (#24301)

fabaff committed:
UPgrade youtube_dl to 2019.05.20 (#24347)

syssi committed:
Add new movement type "actively" of the Xiaomi Vibration Sensor (#24334)

syssi committed:
Fix set_cover_position of the xiaomi_aqara cover for LAN protocol v2 (#24333)

* Fix set_cover_position of the xiaomi_aqara cover for LAN protocol v2 (Closes: #24293)

* Fix lint

michaeldavie committed:
Add Environment Canada weather, sensor, and camera platforms (#21110)

* Added Environment Canada weather platform

* Added Environment Canada weather platform

* Migrate to new folder structure

* Migrate to new folder structure

* Fix updates

* Fix updates again

* Bump env_canada to 0.0.4

* Bump env_canada to 0.0.4

* Bump env_canada to 0.0.4 in requirements_all.txt

* Change daily forecast timestamp and high/low test

* Change daily forecast timestamp and high/low test

* Bump env_canada to 0.0.5

* Break alerts into multiple sensors, bump env_canada to 0.0.6

* Bump env_canada to 0.0.7

* Remove blank line

* Remove 'ec' sensor prefix, bump env_canada to 0.0.8

* Corrections

* Change to manifests.json

* Add docstring to __init.py__

* Update CODEOWNERS

* pylint correction

* pylint correction

* Add alert details, bump env_canada to 0.0.9

* Update requirements_all.txt

* Update .coveragerc

* Bump env_canada to 0.0.10

* Update requirements_all.txt

lufton committed:
Google Cloud Platform component (TTS) (#23629)

* Added Google Cloud TTS service component feature

* Added Neutral voice gender

* Added line break at the end of files

* Updated CODEOWNERS, reqirements_all.txt and .coveragerc

* Fixed some ci/circleci: static-check errors

* Fixed some ci/circleci: static-check error

* Fixed some ci/circleci: pylint errors

* Fixed some ci/circleci: pylint errors

* * made supported_options const
* fixed direct env variable access

* Fixed import order

* * Component renamed
* Added encoding parameter
* Other fixes

* Changed folder name in .coveragerc

* * Removed whitespaces in blank lines
* Split long line

* Removed whitespaces in blank lines

* ci/circleci: static-check

* Fixed requirements_all.txt

* Added speed, pitch and gain parameters

* Added speed, pitch and gain as supported options

* Split too long line

* * Added profiles parameter
* Changed supported languages and encodings values
* Added parameters validations

* Fixes

* Fixes

* Fixes

* Fixes

* Fixes

* Changed options validation

* Added ToggleEntity save and restore state mechanism

* Revert "Added ToggleEntity save and restore state mechanism"

This reverts commit 0e275014

cpopp committed:
Add Streamlabs Water Monitor (#21205)

* Add Streamlabs Water Monitor

* Fail Streamlabswater component setup when given invalid parameters

The Streamlabs Water component is unable to recover if it is given
an invalid API key or location id so this change is to ensure
we validate they are correct during setup and return a failure
if they are not.

* Prime Streamlabswater component sensors so data is available immediately

The sensors for the component were not causing an immediate load of
data from the API when being set up so there was some lag after
startup before values would show up.  This change does an explicit
update when the sensors are setup to ensure data is viewable
immediately after startup.

* Switch Streamlabswater logging to use %s for string formatting

* Update Streamlabswater component with correct dependencies

Dependencies were incorrectly specified using DEPENDS rather
than DEPENDENCIES

* Streamlabswater pull request feedback

Remove detailed class docstrings since they're in the documentation,
reduce code duplication in sensor classes, and remove periods from
the end of log messages.

* Reduce line length in Streamlabswater sensor

* Add docstring on Streamlabswater service callback method

* Get rid of unnecessary initializers in Streamlabswater sensor

* Add manifest file for Streamlabs Water Monitor

* Remove unused REQUIREMENTS

danielkucera committed:
ebusd: added check for monitored conditions validity within correct circuit (#22461)

jkeljo committed:
Bring the Sisyphus integration to silver quality (#22457)

* Bring the Sisyphus integration to silver quality

Checklist:
- [x] (N/A - push integration) Set an appropriate SCAN_INTERVAL (if a polling integration)
- [x] Raise PlatformNotReady if unable to connect during platform setup
- [x] (N/A - no credentials) Handles expiration of auth credentials. Refresh if possible or print correct error and fail setup. If based on a config entry, should trigger a new config entry flow to re-authorize.
- [x] (N/A - local integration) Handles internet unavailable. Log a warning once when unavailable, log once when reconnected.
- [x] Handles device/service unavailable. Log a warning once when unavailable, log once when reconnected.
- [x] Set available property to False if appropriate
- [x] Entities have unique ID (if available)

* Feedback (fix a couple verbose places)

* Use a task instead of a lock

* Initialize field in constructor

* Revert package upgrade.

z0mbieprocess committed:
Support for Salda Smarty XV/XP Ventilation Unit (#21491)

* Support for Salda Smarty XV/XP Ventilation Unit

* Update binary_sensor.py

* Update fan.py

* Update sensor.py

* Update __init__.py

alengwenus committed:
Bump to pypck==0.6.1 (#24356)

tommyjlong committed:
Add templating to MQTT Cover tilt_status (#24355)

fabaff committed:
Upgrade ruamel.yaml to 0.15.97 (#24350)

* Upgrade ruamel.yaml to 0.15.97

* Fix req

bachya committed:
Bump regenmaschine to 1.5.1 (#24358)

pnbruckner committed:
Life360 integration (#24227)

Petro31 committed:
Refactor Waze Travel Time & Update Requirements (#22428)

* Refactor Waze Travel Time & Update Requirements

Refactored Waze Travel Time to contain a data object.

Changed error retrieving data to a warning.
Added distance conversion depending on region.
Removed dependency on TRACKABLE_DOMAINS list.
Update to use WazeRouteCalculator 0.10

3rd time's a charm.  Deleted fork, caused last PR to screw up.  So here we are.

* Update requirements_all.txt

* Revert package upgrade.

* Revert package upgrade.

danielperna84 committed:
Update pyhomematic (#24368)

balloob committed:
Add service to update core location (#24328)

* Add service to update core location

* Update test_init.py

dmulcahey committed:
Fix ZHA battery level when value is reported via signal (#24371)

* fix battery signal
* review comment

dmulcahey committed:
Remove binary sensors for ZHA remotes and controllers (#24370)

* remove remote binary sensor profiles
* fix contact sensors

presslab-us committed:
Add support for ZHA door locks #2 (#24344)

* Add support for DoorLock cluster
* Add test for zha lock
* Change lock_state report to REPORT_CONFIG_IMMEDIATE
* Update channel command wrapper to return the entire result
This allows for return values other than result[1]

* Fix tests
* Fix lint
* Update DoorLock test to work with updated zigpy schema
* Fix lint
* Fix unlock test

pvizeli committed:
Create azure-pipelines-ci.yml

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

Kane610 committed:
Bump dependency (#24376)

Kane610 committed:
Bump dependency (#24376)

fabaff committed:
Upgrade discord.py to 1.1.1 (#24375)

fabaff committed:
Upgrade Mastodon.py to 1.4.3 (#24374)

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

kvanhoorn committed:
Add shuffle support for itunes component (#24319)

* added shuffle support for itunes component

* fixed pylint errors

* more pylint

* final pylint (have my own dev env now)

alengwenus committed:
Fixed wrong setpoint value on startup when climate was previously turned off (#24377)

rolfberkenbosch committed:
Add ATTR_FORECAST_PRECIPITATION option (#24308)

* Add ATTR_FORECAST_PRECIPITATION option

* Remove blank line

Commit:
Provide an option for the DD-WRT device tracker to include non-wireless devices (#24259)

* Use LAN status instead of wireless status for DD-WRT device tracking

* Use the previous DD-WRT device tracker behavior unless specified in the configuration

Commit:
Updated pubnubsub-handler to 1.0.7 to fix crash on slow startup (#24388)

Commit:
Updated pubnubsub-handler to 1.0.7 to fix crash on slow startup (#24388)

pnbruckner committed:
Add for option for template triggers (#24330)

pnbruckner committed:
Improve amcrest error handling and bump amcrest package to 1.5.3 (#24262)

* Improve amcrest error handling and bump amcrest package to 1.5.3

amcrest package update fixes command retry, especially with Digest Authentication, and allows sending snapshot command without channel parameter.

Get rid of persistent_notification.

Errors at startup, other than login errors, are no longer fatal.

Display debug messages about how many times an error has occurred in a row.

Remove initial communications test. If camera is off line at startup this just delays the component setup.

Handle urllib3 errors when getting data from commands that were sent with stream=True.

If errors occur during camera update, try repeating until it works or the camera is determined to be off line.

Drop channel parameter in snapshot command which allows camera to use its default channel, which is different in different camera models and firmware versions.

Make entities unavailable if too many errors occur in a row.

Add new configuration variables to control how many errors in a row should be interpreted as camera being offline, and how frequently to "ping" camera to see when it becomes available again.

Add online binary_sensor option to indicate if camera is available (i.e., responding to commands.)

* Update per review comments

Remove max_errors and recheck_interval configuration variables and used fixed values instead.

Move definition of AmcrestChecker class to module level.

Change should_poll in camera.py to return a fixed value of True and move logic to update method.

Swamp-Ig committed:
Improve boolean validator (#24294)

* Improve boolean validator

* Remove extra throw

* Remove None test as discussed

* Fix for tests depending on None == False

balloob committed:
Add more HomeKit models for discovery (#24391)

* Add more HomeKit models for discovery

* Discover Tradfri with HomeKit

* Add Wemo device info

* Allow full match for HomeKit model

* Fix tests

balloob committed:
Add more HomeKit models for discovery (#24391)

* Add more HomeKit models for discovery

* Discover Tradfri with HomeKit

* Add Wemo device info

* Allow full match for HomeKit model

* Fix tests

balloob committed:
Fix automation failing to restore state (#24390)

* Fix automation off

* Fix tests

balloob committed:
Fix automation failing to restore state (#24390)

* Fix automation off

* Fix tests

balloob committed:
Check cloud trusted proxies (#24395)

balloob committed:
Check cloud trusted proxies (#24395)

balloob committed:
Bumped version to 0.94.1

dmulcahey committed:
add device class to sensors (#24373)

Swamp-Ig committed:
Fix for sun issues (#24309)

Swamp-Ig committed:
Fix for sun issues (#24309)

Kane610 committed:
deCONZ - properly identify configured bridge (#24378)

Kane610 committed:
deCONZ - properly identify configured bridge (#24378)

balloob committed:
Merge pull request #24396 from home-assistant/rc

0.94.1

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Bump version to 0.95.0.dev0

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

frenck committed:
Fixes linter warning in ZHA sensor (#24406)

balloob committed:
Use loose version (#24394)

Kane610 committed:
deCONZ fix retry set state(#24410)

benleb committed:
Merge branch 'add-surepetcare' of github.com:benleb/home-assistant into add-surepetcare

Adminiuga commented:
Perhaps a git rebase is needed. 

> Is it legit that this component is using Python 3.6+ functionalities like f-strings when Hass only requires 3.5.3+ ? I'm just asking because this caused problems at my end

HA Support for Python 3.5 will be dropped after Aug 1st, so until then component should run on Py 3.5 ?

benleb committed:
kick out the whole config flow crap

benleb committed:
remove config flow

amelchio commented:
The PR got messed up from some bad Git voodoo. You are currently proposing 1,595 commits and probably notifying a hundred people. Please close this PR and open a new one.

Don't worry, this happens a lot 😃

benleb commented:
will fix this and also write the docs this weekend! sorry for the delay ✌️

benleb commented:
second edition in #24426
